### DISCUSSION TITLE
Getting  Exception Failed to connect to the database: An exception occurred in the driver: SQLSTATE[08006] [7] error everytime nextcloud does it daily backup

### DISCUSSION THREAD

**Discussion by terrorbyte704** [nextcloud/all-in-one#5067] [Questions] [answered]
### Steps to reproduce
1. let nextcloud do it daily backup or just turn off the container
2.gets about 40 errors about failing to connect to database

### Expected behavior 
when backing up or turning off the containor no error should occur

### Actual behavior 
everytime the containor turns off to backup its gets this error

Exception Failed  to connect to the database: An exception occurred in the driver:  SQLSTATE[08006] [7] could not translate host name  "nextcloud-aio-database" to address: Name does not resolve Exception thrown: Doctrine\DBAL\Exception

### Host OS 
truenas scale, using jailmaker

#### Nextcloud AIO version 
Nextcloud AIO v9.3.0

#### Current channel 
lastest channel

#### Other valuable info 
![image](https://github.com/user-attachments/assets/a8141ee1-8fe1-4104-9b70-7fb14f9c2414)

**Comment by szaimen** [comment]
Hi, can you post the output of `sudo docker info` here?

**Comment by terrorbyte704** [comment]
Client: Docker Engine - Community
 Version:    26.1.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.14.0
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.27.0
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 18
  Running: 15
  Paused: 0
  Stopped: 3
 Images: 24
 Server Version: 26.1.1
 Storage Driver: overlay2
  Backing Filesystem: zfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: e377cd56a71523140ca6ae87e30244719194a521
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.6.32-production+truenas
 Operating System: Debian GNU/Linux 12 (bookworm)
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 46.96GiB
 Name: docker
 ID: dbed4695-a4bb-427e-9a49-22ed3d1a31fb
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

**Comment by szaimen** [comment]
All right. Can you upgrade to docker v27 and recreate the nextcloud-aio network by following https://github.com/nextcloud/all-in-one/discussions/4989 and check if that resolves the problem for you?

**Comment by terrorbyte704** [comment]
Ok i am going to see how to update my docker and will try it out

**Comment by terrorbyte704** [comment]
I have updated my docker and just tried restarting the containor in the aio interface and still got about 50 messages about 

Failed to connect to the database: An exception occurred in the driver: SQLSTATE[08006] [7] could not translate host name "nextcloud-aio-database" to address: Name does not resolve
Exception thrown: Doctrine\DBAL\Exception 

here is my docker info

Client: Docker Engine - Community
 Version:    27.1.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.16.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 17
  Running: 12
  Paused: 0
  Stopped: 5
 Images: 24
 Server Version: 27.1.1
 Storage Driver: overlay2
  Backing Filesystem: zfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 2bf793ef6dc9a18e00cb12efb64355c2c9d5eb41
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.6.32-production+truenas
 Operating System: Debian GNU/Linux 12 (bookworm)
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 46.96GiB
 Name: docker
 ID: dbed4695-a4bb-427e-9a49-22ed3d1a31fb
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

**Comment by szaimen** [comment]
> I have updated my docker and just tried restarting the containor in the aio interface and still got about 50 messages about

Have you also tried to recreate the network as documented in https://github.com/nextcloud/all-in-one/discussions/4989?

**Comment by terrorbyte704** [comment]
just tried it, but still the same error when restarting the containor

**Comment by szaimen** [comment]
So these are new errors and not old ones? Can you show the timestamp?

**Comment by terrorbyte704** [comment]
![image](https://github.com/user-attachments/assets/78718b9b-cbee-447c-8991-4065b48e69e5)

**Comment by szaimen** [comment]
Hm... I fear I still cannot reproduce this on my test server so hard to help further. Maybe someone has an idea at https://help.nextcloud.com?

**Comment by neuhausf** [comment]
Same issue here - I get about 50 messages every day in AIO. I guess the DB isn't available during backup? If this is expected, it should not generate error messages.

**Comment by Serpensin** [comment]
I also got these messages, but the fix is quite easy. It tries to connect to `nextcloud-aio-database`, but that's the name of the docker container and not the hostname. So if you manually edit the database container, just add `nextcloud-aio-database` as it's hostname, and the issues will disappear.
![image](https://github.com/user-attachments/assets/ac2aab8c-0ec2-449a-ae95-29f27173b7e9)
I don't exactly know, how to do this in the terminal, since I use portainer, but I think that's pretty easy to google.

**Comment by sidbena** [comment]
I just had the same issue:
```
Error | cron | ExceptionFailed to connect to the database: An exception occurred in the driver: SQLSTATE[08006] [7] could not translate host name "nextcloud-aio-database" to address: Name does not resolve | 7 Jul 2025, 06:02:43
```

I've also tried setting the database container's hostname to `nextcloud-aio-database` just now, and _so far_ (which is only about 15 minutes, I know) I have no additional related errors.

I still have no (that I know of) pertinent information, but if there's anything I can do to assist in troubleshooting it, just holler!

**Reply by sidbena** [reply]
I see that mine is from `cron` specifically, not sure that means anything in this scenario.

**Comment by woiwoiwoiwoi** [comment]
I also get loads of error messages every day after the backup / container restart. I have not tried changing the hostname yet. Could I cause any damage doing this?

**Reply by Serpensin** [reply]
I changed it a few months ago and nothing happened yet. I guess nothing exploded, because the hostname changes after every container rebuild anyway if you don't set a custom hostname.

**Reply by woiwoiwoiwoi** [reply]
Ah yes, that's a good point, thank you!

**Reply by woiwoiwoiwoi** [reply]
So I manually set the hostname to `nextcloud-aio-nextcloud` (also via Portainer) and the original error message went away. However, now I have a new one:
```
Exception Failed to connect to the database: An exception occurred in the driver:
SQLSTATE[08006] [7] connection to server at "nextcloud-aio-
database" (172.29.96.3), port 5432 failed: Connection refused Is the
server running on that host and accepting TCP/IP connections?
```
I'm guessing it's trying to connect on port 5432, but the database container is probably not yet fully up, so it does not respond? Anybody else having this issue?

**Reply by woiwoiwoiwoi** [reply]
Reading the changelog, AIO 11.3.0 now sets hostnames for all containers. So doing it manually is not required anymore. After the first restart, I did not get any errors. Now, however, I again get it every day after the daily backup routine.

**Comment by orrinwitt** [comment]
I also have this issue, but my question is: is it actually a problem or is it just throrwing errors because the container stack is in the process of starting up? the timestamps are consistently around the same time of day, so I'm guessing that if it were a problem, there would be errors during other times as well?

**Reply by woiwoiwoiwoi** [reply]
My feeling is it probably is not a problem. But the logs being full of this makes it harder to spot actual problems.

**Comment by szaimen** [comment, marked as answer]
Hi, this is should now be fixed with v11.5.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by homonto** [comment]
this still exists on 1st June 2026:

[files_antivirus] Error: OCA\Files_Antivirus\BackgroundJob\BackgroundScanner::processFiles, exception: An exception occurred in the driver: SQLSTATE[08006] [7] could not translate host name "nextcloud-aio-database" to address: Name does not resolve
	from ? by -- at May 28, 2026, 10:27:26 PM

[core] Error: Error while running background job OCA\Files_Antivirus\BackgroundJob\BackgroundScanner (id: 60047643794677761, arguments: null)
	from ? by -- at May 28, 2026, 10:27:26 PM

[cron] Error: Failed to connect to the database: An exception occurred in the driver: SQLSTATE[08006] [7] could not translate host name "nextcloud-aio-database" to address: Name does not resolve
	from ? by -- at May 28, 2026, 10:27:26 PM

**Reply by Hyperadministrator** [reply]
Same issue here at 07.07.2026. It started after updating NextCloud master container to latest version today and also running OS updates after that in Ubuntu server.
